controllers/storagecluster/storagequota_test.go

```diff
@@ -0,0 +1,168 @@
+package storagecluster
+
+import (
+	"context"
+	"testing"
+
+	configv1 "github.com/openshift/api/config/v1"
+	quotav1 "github.com/openshift/api/quota/v1"
+	fakequota "github.com/openshift/client-go/quota/clientset/versioned/fake"
+	api "github.com/openshift/ocs-operator/api/v1"
+	"github.com/stretchr/testify/assert"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	k8sVersion "k8s.io/apimachinery/pkg/version"
+	"sigs.k8s.io/controller-runtime/pkg/client/fake"
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
+)
+
+var testQuantity1T = resource.MustParse("1Ti")
+var testQuantity2T = resource.MustParse("2Ti")
+var testStorageClusterWithOverprovision = &api.StorageCluster{
```

Comment:
Rather than defining a new StorageCluster, define a valid StorageDeviceSets list with appropriate OPCs. Since you already have a function for the DeepCopy, you could do a DeepCopy of the existing mock StorageCluster and then DeepCopy the SDSs/OPCs into that.